packages/server/src/auth/CrossmintAuthServer.ts

```diff
@@ -58,12 +70,78 @@ export class CrossmintAuthServer extends CrossmintAuth {
         return user;
     }
 
+    public async handleCustomRefresh(request: GenericRequest, response?: ServerResponse): Promise<GenericResponse> {
+        // If the request from the client includes a refresh token (like the OneTimeSecret after authenticating), use that
+        // Otherwise, try to get the refresh token from the cookies
+        const body = isFetchRequest(request) ? await request.json() : await getNodeRequestBody(request);
```

Comment:
The `isFetch` / `isNode` conditionals are all over the place.
Maybe we can have two separate classes with a shared interface, and select the correct class once at the beginning of this method?
Something like 

```typescript
interface RequestAdapter {
  toJson();
  toResponseError(message);
  withHeaders(headers): RequestAdapter;
  withBody(body): RequestAdapter;
  // etc.
}

class FetchRequestAdapter implements RequestAdapter {
    // fetch-specific stuff here
}
class NodeRequestAdapter implements RequestAdapter {
     // node-specific stuff here
}

Reply:
I actually only do it there and in the cookies util to fetch the header so I think this abstraction is not worth adding right now as I think the current code is pretty declarative to read. there is one other `isFetchRequest` but that is only to prepare the appropriate response. wdyt?

Reply:
I don't know, I don't feel that this code is "pretty declarative" at all tbh.
There's 5 uses of either `isFetch` or `isNode` just in this method, and you're saying there's other uses outside of it...